code/webauthn/contracts/contracts/Account.sol

```diff
@@ -0,0 +1,356 @@
+        // vice versa. If your library also generates signatures with 0/1 for v instead 27/28, add 27 to v to accept
+        // these malleable signatures as well.
+        if (
+            uint256(s) >
+            0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A0
+        ) {
+            magic = bytes4(0);
+        }
+
+        address recoveredAddress = ecrecover(_hash, v, r, s);
+
+        // Note, that we should abstain from using the require here in order to allow for fee estimation to work
+        if (recoveredAddress != owner) {
+            magic = bytes4(0);
+        }
+        if(recoveredAddress == address(0)){
+            magic = bytes4(0);
+        }
+    }
+
+    function payForTransaction(
+        bytes32,
+        bytes32,
+        Transaction calldata _transaction
+    ) external payable override onlyBootloader {
+        bool success = _transaction.payToTheBootloader();
+        require(success, "Failed to pay the fee to the operator");
+    }
+
+    function prepareForPaymaster(
+        bytes32, // _txHash
+        bytes32, // _suggestedSignedHash
+        Transaction calldata _transaction
+    ) external payable override onlyBootloader {
+        _transaction.processPaymasterInput();
+    }
+
+     function _decodeWebauthnSignature(
+        bytes memory webauthnSignature
+    )
+        private
+        pure
+        returns (
+            bytes memory authenticatorData,
+            bytes memory clientData,
+            bytes32[2] memory rs,
+            bytes32 signedMessage
+        )
+    {
+        (authenticatorData, 
+        clientData, 
+        rs,
+        signedMessage
+        ) = abi.decode(
+            webauthnSignature,
+            (bytes, 
+            bytes, 
+            bytes32[2],
+            bytes32
+            )
+        );
+      
+    }
+
+    function validateWebAuthnSignature(
+        // bytes32 challenge,
+        bytes memory webauthnSignature,
+        bytes32 txHash
+    ) private view returns (bool valid) {
+        if(r1Owner.length == 0){
+            return false;
+        }
+        bytes32[2] memory pubKey = abi.decode(r1Owner, (bytes32[2]));
+        valid = _validateWebAuthnSignature(txHash, webauthnSignature, pubKey);
+    }
+
+
+      function _validateWebAuthnSignature(
+        // bytes32 challenge,
+        bytes32 txHash,
+        bytes memory webauthnSignature,
+        bytes32[2] memory pubKey
+    ) private view returns (bool valid) {
+        (
+            bytes memory authenticatorData,
+            bytes memory clientData,
+            bytes32[2] memory rs,
+            bytes32 signedMessage
+        ) = _decodeWebauthnSignature(webauthnSignature);
+
+        // malleability check
+        if (rs[1] > lowSmax) {
+            return false;
+        }
+
+        // check if the signed message is the same as the transaction hash
+        if(txHash != signedMessage){
```

Comment:
`signedMessage` is extracted from the `webauthnSignature` that is pure user input, the real message that is verified is `message` that is not connected to transaction hash anyhow